woocommerce/woocommerce-ios
# Edit Products: show product reviews on main screen

Part of #2082

## Description
In this PR I integrated the Reviews cell in our new Product Detail screen, under the Price cell. After some investigations, it seems that reviews are visible for all the product types, so we will show the cell in all the products that actually we support in the app (Simple, Affiliate, Grouped, Variable). This PR introduces only the cell visible in the Product Detail, the action will be part of another PR.

## Changes
- New `productReviewsImage` image.
- Renamed the existing `ProductReviewsTableViewCell` to `OldProductReviewsTableViewCell` (it's still used in the old Product Detail screen, that in the future will be removed).
- Updated `ProductFormActionsFactory` to support reviews.
- New `ProductReviewsTableViewCell` with a new appearance.
- Updated `DefaultProductFormTableViewModel`  with the reviews row.
- Updated `ProductFormDataModel` with the reviews fields: `averageRating` and `ratingCount `.
- Updated `ProductFormSection+ReusableTableRow` with the reviews row.
- Updated `ProductFormTableViewDataSource` that handle the reviews cell configuration.
- Added the action to handle reviews in `ProductFormViewController`.
- `ProductFormViewModel` and `ProductFormViewModelProtocol` update reviews.
- Updated tests.

## Testing
- Navigate to a product detail with some reviews.
- Make sure that you are able to see the reviews cell populated.
- Try the same process with different products type (Simple, Affiliate, Grouped, Variable) and with Products without reviews.

## Screenshots
| 0 Reviews             | Populated reviews | Reviews with dark mode enabled |
:-------------------------:|:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 11 Pro - 2020-08-04 at 12 36 14](https://user-images.githubusercontent.com/495617/89284462-1e758480-d64f-11ea-9e16-a76567eaf759.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-08-04 at 13 05 15](https://user-images.githubusercontent.com/495617/89286840-2d5e3600-d653-11ea-958d-1a2e97922aca.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-08-04 at 12 31 48](https://user-images.githubusercontent.com/495617/89284097-87103180-d64e-11ea-9337-42d6250b5155.png)







Update release notes:
